services/tone_analyzer/v3-beta.js

```diff
@@ -62,32 +62,49 @@ module.exports = function (RED) {
         version_date: '2016-02-11'
       });
 
-      var options = {
+      var hasJSONmethod = (typeof msg.payload.toJSON === 'function') ;
+      var isBuffer = false;
+      if (hasJSONmethod==true)
+      {
+        if (msg.payload.toJSON().type == 'Buffer')
+            isBuffer=true;
+      }
+
+      // Payload (text to be analysed) must be a string (content is either raw string or Buffer)
+      if (typeof msg.payload == 'string' ||  isBuffer == true )
+      {
+         var options = {
         text: msg.payload,
         sentences: config.sentences
-      };
-
-      if (config.tones !== 'all') {
-        options.tones = config.tones;
-      }
+        };
 
-      node.status({fill:"blue", shape:"dot", text:"requesting"});
-      tone_analyzer.tone(options, function (err, response) {
-        node.status({})
-        if (err) {
-          node.error(err, msg);
-        } else {
-          msg.response = response;
+        if (config.tones !== 'all') {
+          options.tones = config.tones;
         }
 
-        node.send(msg);
-      });
+        node.status({fill:'blue', shape:'dot', text:'requesting'});
+        tone_analyzer.tone(options, function (err, response) {
+          node.status({})
+          if (err) {
+            node.error(err, msg);
+          } else {
+            msg.response = response;
+          }
+
+          node.send(msg);
+        });
+      }
+      else {
+        var message = 'The payload must be either a string or a Buffer';
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["message" is already defined](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2393366918/issues/source?bid=3225115&fileBranchId=3230140#l98)